### PR TITLE
Updated initialisation of `Color` values and `std.Progress` nodes to respect environment variable `NO_COLOR` and `native_os`.

### PR DESCRIPTION
**Pull request by amp-59**

### CODE REVIEW

**Comment by amp-59** [comment]
@Vexu Is this patch ready to be merged?

File: src/main.zig [hunk 1]
```diff
@@ -4405,7 +4410,7 @@ fn cmdTranslateC(
     prog_node: std.Progress.Node,
 ) !void {
     if (build_options.only_core_functionality) @panic("@translate-c is not available in a zig2.c build");
-    const color: Color = .auto;
+    const color: Color = if (native_os == .wasi or EnvVar.NO_COLOR.isSet()) .off else .auto;
```
Comment (Vexu) [inline src/main.zig:4413]: `cmdTranslateC` should get `color` from either `buildOutputType` or `serve`.

**Review by Vexu — COMMENTED** [review]
It would be nice having the comment explaining the initializer attached to these other places, maybe by extracting the initializer into a `defaultColor` helper?

**Comment by amp-59** [comment]
Is that all okay?

**Comment by Vexu** [comment]
If you fix the commits then yes. `defaultColor` didn't need to be more verbose but I guess it's fine that way too.

**Review by andrewrk — CHANGES_REQUESTED** [review]
This diff has a bunch of crap in it that is already in master branch. If you want this reviewed then make it mergeable as-is.

**Comment by amp-59** [comment]
>  If you want this reviewed 

No problem.

**Pull request closed without merging**